lib/index.js

```diff
@@ -121,12 +121,19 @@ function Socket (type, opts) {
   }
 
   /* start listening for inbound messages */
-  if(this.af_domain === nn.AF_SP) {
-    if (this.receiver) this._startPollReceive();
+  if (this.af_domain === nn.AF_SP && this.receiver) {
+    this._startPollReceive();
   }
+
+  if (!opts.hasOwnProperty('objectMode')) {
+    opts.objectMode = true;
+  }
```

Comment:
no you configure it from highWaterMark option.

https://iojs.org/api/stream.html#stream_new_stream_readable_options
or
https://nodejs.org/api/stream.html#stream_new_stream_readable_options